core/errors.go

```diff
@@ -0,0 +1,29 @@
+package core
+
+import "strconv"
+
+// Error defines the core logic error to facilitate errors handling across the product elements.
+type Error struct {
+	Service                   string
+	Stage                     string
+	Message                   string
+	ServiceResponseStatusCode int
+}
+
+func (e Error) Error() string {
+	return "[" + e.Service + "][stage:" + e.Stage + "][http code: " + strconv.Itoa(e.ServiceResponseStatusCode) + "] " +
```

Comment:
**suggestion** why not tag the errors fields with `json` and printing out the error as a JSON as follows:
```
func (e Error) Error() string {
	ret, _ := json.Marshal(e)
	return string(ret)
}
```

This way the error will be aggregated by CloudWatch as a structured one and might be easier to debug and filter.

Reply:
Could you please highlight the benefits of your proposal?

My rationale is straightforward: if to do something custom, try doing it simple - with least dependencies - as soon as it does not pose the maintenance costs higher compared to the solution with dependencies.